Lib/enum.py

```diff
@@ -731,14 +731,16 @@ def __contains__(cls, value):
         """
         if isinstance(value, cls):
             return True
-        try:
-            cls(value)
-            return True
-        except ValueError:
-            return (
-                    value in cls._unhashable_values_    # both structures are lists
-                    or value in cls._hashable_values_
-                    )
+        if issubclass(cls, Flag):
+            try:
+                result = cls._missing_(value)
+                return isinstance(result, cls)
+            except ValueError:
+                pass
+        return (
+                value in cls._unhashable_values_    # both structures are lists
+                or value in cls._hashable_values_
```

Comment:
Would this check be sufficient for other enums? shouldn't we once again check for `_value2member_map_` as well?

Reply:
If it's in `_value2member_map_`, it's also in `_hashable_values_`.